wagtail/search/backends/database/mysql/mysql.py

```diff
@@ -0,0 +1,587 @@
+            #
+            #   Or(Not(Term("A")), Not(Term("B")))
+            #
+            # It's simpler to code it this way as we only need to store the
+            # invert status of the terms rather than all the operators.
+
+            subquery_lexemes = [
+                self.build_search_query_content(subquery, invert=invert)
+                for subquery in query.subqueries
+            ]
+
+            is_and = isinstance(query, And)
+
+            if invert:
+                is_and = not is_and
+
+            if is_and:
+                return balanced_reduce(lambda a, b: a & b, subquery_lexemes)
+            else:
+                return balanced_reduce(lambda a, b: a | b, subquery_lexemes)
+
+        raise NotImplementedError(
+            '`%s` is not supported by the MySQL search backend.'
+            % query.__class__.__name__)
+
+    def build_search_query(self, query):
+        return self.build_search_query_content(query)
+
+    def get_index_vectors(self, search_query):
+        return [
+            (F('index_entries__title'), F('index_entries__title_norm')),
+            (F('index_entries__body'), 1.0),
+        ]
+
+    def get_fields_vectors(self, search_query):
+        raise NotImplementedError()
+
+    def get_search_vectors(self, search_query):
+        if self.fields is None:
+            return self.get_index_vectors(search_query)
+
+        else:
+            return self.get_fields_vectors(search_query)
+
+    def _build_rank_expression(self, vectors, config):
+        rank_expressions = [
+            self.build_tsrank(vector, self.query, config=config) * boost
+            for vector, boost in vectors
+        ]
+
+        rank_expression = rank_expressions[0]
+        for other_rank_expression in rank_expressions[1:]:
+            rank_expression += other_rank_expression
+
+        return rank_expression
+
+    def search(self, config, start, stop, score_field=None):
+        # TODO: Handle MatchAll nested inside other search query classes.
+        if isinstance(self.query, MatchAll):
+            return self.queryset[start:stop]
+
+        elif isinstance(self.query, Not) and isinstance(self.query.subquery, MatchAll):
+            return self.queryset.none()
+
+        if isinstance(self.query, Not):  # If the outermost operator is a Not, we invert the query. This is done because, if every search term is negated, the Not() will return no results, an we want to match all results instead.
+            query = self.query.subquery
+            negated = True
+        else:
+            query = self.query
+            negated = False
+
+        search_query = self.build_search_query(query)
+        match_expression = MatchExpression(search_query, output_field=BooleanField())  # For example: MATCH (`title`, `body`) AGAINST ('+query' IN BOOLEAN MODE)
+        score_expression = MatchExpression(search_query, output_field=FloatField())  # Same as above, but with a different output_field
```

Comment:
Looks like ``title_norm`` isn't being used. Is it possible to create a separate ``MatchExpression`` for `body` and `title` and multiply the `title` by `title_norm`?

in the postgres backend, it's done here: https://github.com/wagtail/wagtail/blob/1b50c744240b1f6ddaf5abd99c2ea26d17c5cdfb/wagtail/contrib/postgres_search/backend.py#L494


As a total guess (sorry if this is wrong!), but do you think something like this would work?

```python
score_expression = (
    MatchExpression(search_query, columns=['title'], output_field=FloatField()) * F('title_norm') +
    MatchExpression(search_query, columns=['body'], output_field=FloatField())
)
```